File: frontend/src/app/shared/models/payment.ts
export interface PaymentResponse {
  paymentId: number;
  orderId: number | null;
  paymentMethod: string | null;
  paymentStatus: string | null;
  paidAt: string | null;
}

export interface CreatePaymentRequest {
  orderId?: number;
  paymentMethod: string;
  paymentStatus: string;
}

export interface UpdatePaymentRequest {
  orderId?: number;
  paymentMethod: string;
  paymentStatus: string;
  paidAt?: string | null;
}

File: backend/Models/Payment.cs
using System;
using System.Collections.Generic;

namespace backend.Models;

public partial class Payment
{
    public int PaymentId { get; set; }

    public int? OrderId { get; set; }

    public string? PaymentMethod { get; set; }

    public string? PaymentStatus { get; set; }

    public DateTime? PaidAt { get; set; }

    public virtual Order? Order { get; set; }
}
